packages/app/src/heartbeatService.ts

```diff
@@ -117,34 +119,41 @@ export class HeartbeatServiceImpl implements HeartbeatService {
    * Returns a base64 encoded string which can be attached to the heartbeat-specific header directly.
    * It also clears all heartbeats from memory as well as in IndexedDB.
    *
-   * NOTE: It will read heartbeats from the heartbeatsCache, instead of from indexedDB to reduce latency
+   * NOTE: Consuming product SDKs should not send the header if this method
+   * returns an empty string.
    */
   async getHeartbeatsHeader(): Promise<string> {
     if (this._heartbeatsCache === null) {
       await this._heartbeatsCachePromise;
     }
-    // If it's still null, it's been cleared and has not been repopulated.
-    if (this._heartbeatsCache === null) {
+    // If it's still null or the array is empty, there is no data to send.
+    if (
+      this._heartbeatsCache === null ||
+      this._heartbeatsCache.heartbeats.length === 0
```

Comment:
is it possible for `this._heartbeatsCache.heartbeats` to be undefined while `this._heartbeatsCache` is not null.

Might be worth an additional check on `heartbeats`

Reply:
Seeing below it's not super obvious how it's even possible

Reply:
Typescript should catch if there's ever an attempt to assign an object to this._heartbeatsCache that doesn't have a heartbeats property.